core/src/consensus.rs

```diff
@@ -499,7 +503,6 @@ impl Tower {
                                 // is an ancestor of the current root, because `candidate_slot` is a
                                 // descendant of the current root
                                 && *lockout_interval_start > root
-                                && !locked_out_vote_accounts.contains(vote_account_pubkey)
```

Comment:
IMO, another addition of chained conditions increases human's cognitive load exponentially. So, I wanted to separate one for the looping construct out of this `if`. This `if` should only be comprised of business logic.